### PR TITLE
DCOS_OSS-1050: Remove image if no id is provided

### PR DESCRIPTION
**Pull request by Poltergeist**
Remove image if a container id has been removed
but was available earlier.

Closes DCOS_OSS-1050

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

File: plugins/services/src/js/reducers/serviceForm/Containers.js [hunk 1]
```diff
@@ -539,6 +539,9 @@ module.exports = {
       newState[index] = Object.assign({}, newState[index], {
         image: { id: value, kind: "DOCKER" }
       });
+      if (value === "") {
```
Comment (orlandohohmeier) [inline plugins/services/src/js/reducers/serviceForm/Containers.js]: Would love if we could get away without adding another condition... #style

**Comment by orlandohohmeier** [comment]
I'm ok with merging this one as is, but would love if you could check if there is away to not introduce another condition.

File: plugins/services/src/js/reducers/serviceForm/Containers.js (hunk 1 above)
Comment (MatApple) [inline plugins/services/src/js/reducers/serviceForm/Containers.js]: can we move condition to outer condition @Poltergeist?
```
if (type === SET && joinedPath === `containers.${index}.image.id` && value !== "") {
```

Comment (Poltergeist) [inline plugins/services/src/js/reducers/serviceForm/Containers.js]: Problem is that we operate on the last state, so we need to remove the image attribute if it is not set, so I fear we are not able to remove the condition. Except, we always call the delete before we do the object assign.

Comment (nLight) [inline plugins/services/src/js/reducers/serviceForm/Containers.js]: can't we do:
 
```javascript
const image = value === ""
  ? {}
  : {image: { id: value, kind: "DOCKER" }};

newState[index] = Object.assign({}, newState[index], image);
```

Comment (Poltergeist) [inline plugins/services/src/js/reducers/serviceForm/Containers.js]: We would still have image: `null` in the resulting object.

**Pull request merged**